validators/generate_uris.py

```diff
@@ -139,27 +141,26 @@ def process_document(data):
     merged_string = ''
     if 'doi' in data:
         merged_string = merged_string + custom_slugify(data['doi'])
-    merged_string += '-'
-
-    if 'uri' in data:
+        merged_string += '-'
+    elif 'uri' in data:
         merged_string = merged_string + custom_slugify(data['uri'])
-    merged_string += '-'
-
-    if 'title' in data:
-        merged_string = merged_string+ custom_slugify(data['title'])
-    merged_string += '-'
```

Comment:
doi and uri are two unique properties, if one of them present, IMO, we should only use it to keep the URI short

Reply:
That is what I changed. However, I have also kept the option when the data does not have either to merge all the other properties.